hilbert/scheduler.py

```diff
@@ -31,23 +29,78 @@ def step(self):
             self.pointer += 1
 
 
-class LinearLRScheduler:
+class LearningRateScheduler:
+    def __init__(self,
+                 optimizer,
+                 start_lr,
+                 num_epochs,
+                 end_lr=0):
+        """
+        Base class for all learning rate schedulers.
+
+        :param optimizer: Optimizer used for training objective.
+        :param start_lr: Initial learning rate.
+        :param num_epochs: Number of epochs
+        :param end_lr: Stationary state learning rate
+        """
 
-    def __init__(self, optimizer, start_lr, num_epochs, end_lr=0):
         self.opt = optimizer
         self.start_lr = start_lr
         self.end_lr = end_lr
         self.num_epochs = num_epochs
         self.cur_epoch = -1
         self.step()
 
+    def step(self):
+        raise NotImplementedError('Not implemented!')
+
+
+class LinearLRScheduler(LearningRateScheduler):
+
+    def __init__(self, optimizer, start_lr, num_epochs, end_lr=0):
+        '''
+
+        :param optimizer: Optimizer used for training objective.
+        :param start_lr: Initial learning rate.
+        :param num_epochs: Number of epochs
+        :param end_lr: Stationary state learning rate
+        '''
+        super(LinearLRScheduler, self).__init__(optimizer,
+                                                start_lr,
+                                                num_epochs,
+                                                end_lr)
+
     def step(self):
         self.cur_epoch += 1
         if self.cur_epoch < self.num_epochs:
             fraction_left = 1 - self.cur_epoch / self.num_epochs
             cur_lr = self.end_lr+(self.start_lr-self.end_lr)*fraction_left
         else:
             cur_lr = self.end_lr
+
+        for param_group in self.opt.param_groups:
+            param_group['lr'] = cur_lr
+
+
+class InverseLRScheduler(LearningRateScheduler):
+    def __init__(self, optimizer, start_lr, num_epochs):
+        """
+        Based on the learning rate scheduler O(1/t) from Bergstra and Bengio (2012), where keeping learning rate
+        constant for the first [num_epochs] updates.
+        cur_lr = start_lr * num_epoch/max(t, num_epoch)
+
+        :param optimizer: Optimizer used for training objective.
+        :param start_LR: Initial learning rate.
+        :param num_epoch: Number of epochs keeping learning rate constant
+        """
+
+        super(InverseLRScheduler, self).__init__(optimizer,
```

Comment:
Minimal indent style
```suggestion
        super(InverseLRScheduler, self).__init__(
            optimizer,
```